safesocket/safesocket_darwin.go

```diff
@@ -46,6 +47,15 @@ func localTCPPortAndTokenMacsys() (port int, token string, err error) {
 	if auth == "" {
 		return 0, "", errors.New("empty auth token in sameuserproof file")
 	}
+
+	// The above files exist forever after the first run. Check that the server
+	// is actually running (though it may still be in a Stopped state).
```

Comment:
I'd suggest rephrasing the comment a bit to reflect that we're checking that _some_ server is listening on the port, not that something is running. Perhaps "avoid returning a port that nothing is listening on"?

Also, I was surprised to see `127.0.0.1` here rather than `localhost` and tracked it down to #7851. Could you please also add a short comment referencing that issue?

Reply:
Clarified both points in https://github.com/tailscale/tailscale/pull/9234/commits/c27afd755fcd81345844082427410c62b9f9607d.